src/Types/Solver.hs

```diff
@@ -49,7 +50,8 @@ data SolverState = SolverState {
     _instanceCounts :: HashMap Id Int, -- Number of instantiations for a real-name, used in selecting representative
     _toRemove :: [Id],
     _useCount :: Map Id Int,
-    _messageChan :: Chan Message
+    _messageChan :: Chan Message,
+    _sampleResult :: SampleResult
```

Comment:
I don't think the test-filter's state should be entangled with the Petrinet solver state. Since later we plan to decouple the solver from the test-filter, I think you should try to keep them separate as much as possible. Perhaps a nested StateT monad would be easy enough. Let me know how that goes